Composer/packages/client/src/pages/design/VisualEditor.tsx

```diff
@@ -81,8 +81,15 @@ const VisualEditor: React.FC<VisualEditorProps> = (props) => {
     // Override the sdk.schema 'title' field with form ui option 'label' field
     // to make sure the title is consistent with Form Editor.
     Object.entries(formConfig).forEach(([$kind, formOptions]) => {
-      if (formOptions && sdkDefinitions[$kind]) {
-        sdkDefinitions[$kind].title = formOptions?.label;
+      const formLabel = formOptions?.label;
+      const definition = sdkDefinitions[$kind];
+      if (formLabel && definition) {
+        if (typeof formLabel === 'string') {
+          definition.title = formLabel;
+        } else if (typeof formLabel === 'function') {
+          const computedTitle = formLabel({});
+          if (computedTitle) definition.title = computedTitle;
+        }
```

Comment:
Could look better if we use switch instead of if else

Reply:
We may also catch the "default" case or the "else" case with a value saying "unknown"

Reply:
I moved the type judgement logic inside title resolver considering that the label fn needs to be aware of data context.

Reply:
it can resolve the 'else' case